keyboards/thevankeyboards/minivan/keymaps/josjoha/keymap.c

```diff
@@ -0,0 +1,1187 @@
+
+
+        /*      Left Shift layer to ...
+         *
+         * Uncomment one of the below lines, determining where L-shift tap-toggles to on the 
+         * BASE layer. 
+         */
+//#define LSHIFT_LAYER_RAR // Be warned and don't hold it against me if you accidentally hit 'Power' at the wrong moment.
+//#define LSHIFT_LAYER_MOV // Handy to have navigation on a toggle. 
+                         // _MOV is the least dangerous layer to accidentally activate.
+//#define LSHIFT_LAYER_DRA // _DRA is also the least easy to access layer normally, on pinky which is sortof wrong.
+                         // This would help alleviate it.
+//#define LSHIFT_LAYER_ACC // If typing a lot of these in a row
+#define LSHIFT_LAYER_PAD // Easier Access to numpad (for default shortcuts in blender(1) for example).
+
+
+        /*      Sacrificing functionality when compiled hex file is too large
+         *
+         * You can cut some parts of the functionality, if device space is too small.
+         * The overall QMK code sometimes changes, which causes the firmware to change
+         * size. See also in config.h about space saving #defines.
+         *
+         * Default (and how documentation represents the keymap): 
+         *           - all undefined
+         */
+//#define space_cut_descr_arrow_ud // Uncomment to remove up/down arrows in the 'descramble mode' Unicode layer.
+                                 // This saves a couple of dozen bytes.
+//#define space_cut_normal_arrow_ud // Uncomment to remove up/down arrows in the 'normal mode' Unicode layers.
+                                  // This saves only a few bytes, but harmonizes the keymap if 'descramble' arrows cut.
+//#define space_cut_descr_dashes // Uncomment to remove dashed lines in the 'descramble mode' Unicode layer.
+                               // This saves a couple of dozen bytes.
+
+// Below here no more comfortable configuration options.....
+// --------------------------------------^---------------------------------------
+
+  /* Overview of the code
+   *
+   * The two compile options imply that the Qwerty-Dvorak variant re-defines
+   * the two letter and the two number/symbol layers, while not compiling the
+   * special Linux Unicode 'descramble' layers. The Qwerty-Dvorak letters number/
+   * symbol layers are located in qwerty_dvorak.c. There is only one #define
+   * dealing with this. The unicode macros needed for the special Linux Unicode
+   * are also not compiled in the Qwerty-Dvorak version.
+   *
+   * The Unicode is in unicode_macros.c/h.
+   */
+
+
+// Set up user GUI choice:
+#ifndef SWITCH_GUIS
+    #define KC__XGUI KC_LGUI
+    #define KC__YGUI KC_RGUI
+#endif
+#ifdef SWITCH_GUIS
+    #define KC__XGUI KC_RGUI
+    #define KC__YGUI KC_LGUI
+#endif
+
+
+// Some things from broader QMK code:
+#include QMK_KEYBOARD_H
+#define RGBLIGHT_SLEEP // QMK docs: "If defined, the RGB lighting will be switched off when the host goes to sleep"
```

Comment:
This needs to be defined in the config.h, or it will not be picked up correctly.

Reply:
Ok, I left #include QMK_KEYBOARD_H in keymap.c because I got errors moving it to config.h, and moved #define RGBLIGHT_SLEEP to config.h, hoping I understood you correctly.

Reply:
Yup, that's correct